tests/test_topics.py

```diff
@@ -738,6 +738,96 @@ def foo():
 
         asyncio.get_event_loop().run_until_complete(doit())
 
+    def test_controller_command_success(self):
+        """Test ack when a controller command succeeds.
+        """
+        async def doit():
+            async with Harness(initial_state=salobj.State.ENABLED) as harness:
+
+                ackcmd = await harness.remote.cmd_wait.set_start(duration=0, timeout=STD_TIMEOUT)
+                self.assertEqual(ackcmd.ack, salobj.SalRetCode.CMD_COMPLETE)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_controller_command_return_ackcmd(self):
+        """Test ack when controller command callback returns an ackcmd.
+        """
+        async def doit():
+            async with Harness(initial_state=salobj.State.ENABLED) as harness:
+                ack = salobj.SalRetCode.CMD_NOPERM
+
+                def return_ack(data):
+                    return harness.csc.salinfo.makeAckCmd(private_seqNum=data.private_seqNum,
+                                                          ack=salobj.SalRetCode.CMD_NOPERM)
+
+                harness.csc.cmd_wait.callback = return_ack
+                with salobj.assertRaisesAckError(ack=ack):
+                    await harness.remote.cmd_wait.start(timeout=STD_TIMEOUT)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_controller_command_expected_error(self):
+        """Test ack when controller command callback raises ExpectedError.
+        """
+        async def doit():
+            async with Harness(initial_state=salobj.State.ENABLED) as harness:
+                msg = "intentional failure"
+
+                def fail_expected_exception(data):
+                    raise salobj.ExpectedError(msg)
+
+                harness.csc.cmd_wait.callback = fail_expected_exception
+                with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED,
+                                                 result_contains=msg):
+                    await harness.remote.cmd_wait.start(timeout=STD_TIMEOUT)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_controller_command_exception(self):
+        """Test ack when controller command callback raises an Exception.
```

Comment:
I think my problem with these docstrings is that I'm having trouble parsing "Test ack..." My brain tries to ask "Test ack what?" Maybe "Test the `ack` value in the raised exception when ..." instead (assuming I parsed that right)?

Reply:
Fixed above. I hope.